src/toil/test/server/serverTest.py

```diff
@@ -29,35 +31,228 @@
+            # Should allow clearing
+            store.set('id1', 'key1', None)
+            self.assertEqual(store.get('id1', 'key1'), None)
+            self.assertEqual(store.get('id2', 'key1'), 'value2')
+            self.assertEqual(store.get('id1', 'key2'), 'value3')
+
+            store.set('id2', 'key1', None)
+            store.set('id1', 'key2', None)
+            self.assertEqual(store.get('id1', 'key1'), None)
+            self.assertEqual(store.get('id2', 'key1'), None)
+            self.assertEqual(store.get('id1', 'key2'), None)
+
+class FileStateStoreTest(hidden.AbstractStateStoreTest):
+    """
+    Test file-based state storage.
+    """
+
+    from toil.server.utils import AbstractStateStore
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.state_store_dir = self._createTempDir()
+
+    def get_state_store(self) -> AbstractStateStore:
+        """
+        Make a state store to test, on a single fixed local path.
+        """
+
+        from toil.server.utils import FileStateStore
+
+        return FileStateStore(self.state_store_dir)
+
+class FileStateStoreURLTest(hidden.AbstractStateStoreTest):
+    """
+    Test file-based state storage using URLs instead of local paths.
+    """
+
+    from toil.server.utils import AbstractStateStore
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.state_store_dir = 'file://' + self._createTempDir()
+
+    def get_state_store(self) -> AbstractStateStore:
+        """
+        Make a state store to test, on a single fixed URL.
+        """
+
+        from toil.server.utils import FileStateStore
+
+        return FileStateStore(self.state_store_dir)
+
+@needs_aws_s3
+class AWSStateStoreTest(hidden.AbstractStateStoreTest):
+    """
+    Test AWS-based state storage.
+    """
+
+    from toil.server.utils import AbstractStateStore
+
+    from mypy_boto3_s3 import S3ServiceResource
+    from mypy_boto3_s3.service_resource import Bucket
+    from toil.jobStores.aws.jobStore import AWSJobStore
+    from toil.lib.aws import get_current_aws_region
+    from toil.lib.aws.utils import create_s3_bucket, delete_s3_bucket
+    from toil.lib.aws import session
+
+    region: Optional[str]
+    s3_resource: Optional[S3ServiceResource]
+    bucket: Optional[Bucket]
+    bucket_name: Optional[str]
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+        Set up the class with a single pre-existing AWS bucket for all tests.
+        """
+        super().setUpClass()
+
+        cls.region = get_current_aws_region()
+        cls.s3_resource = session.resource("s3", region_name=cls.region)
+
+        cls.bucket_name = f"toil-test-{uuid.uuid4()}"
+        cls.bucket = create_s3_bucket(s3_resource, bucket_name, cls.region)
+        cls.bucket.wait_until_exists()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        if cls.bucket_name:
+            delete_s3_bucket(cls.s3_resource, cls.bucket_name, cls.region)
+        super().tearDownClass()
```

Comment:
Since the imports are imported as class variables, I think you need `cls.` to access them. Or, I also like the `HAVE_S3` in `server/utils.py`, so perhaps we could do something similar here as well?

```suggestion
    @classmethod
    def setUpClass(cls) -> None:
        """
        Set up the class with a single pre-existing AWS bucket for all tests.
        """
        super().setUpClass()

        cls.region = cls.get_current_aws_region()
        cls.s3_resource = cls.session.resource("s3", region_name=cls.region)

        cls.bucket_name = f"toil-test-{uuid.uuid4()}"
        cls.bucket = cls.create_s3_bucket(cls.s3_resource, cls.bucket_name, cls.region)
        cls.bucket.wait_until_exists()

    @classmethod
    def tearDownClass(cls) -> None:
        if cls.bucket_name:
            cls.delete_s3_bucket(cls.s3_resource, cls.bucket_name, cls.region)
        super().tearDownClass()
```

Reply:
I'll just import these where they are used instead. I am missing some `cls`es on the variables though. And apparently these tests don't actually get hit.